compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt

```diff
@@ -97,11 +97,18 @@ internal abstract class BaseComposeScene(
 
     @Volatile
     private var hasPendingDraws = true
+    private var hasPostponedInvalidations = false
     protected fun updateInvalidations() {
         hasPendingDraws = frameClock.hasAwaiters ||
-            snapshotInvalidationTracker.hasInvalidations
-        if (hasPendingDraws && !isInvalidationDisabled && !isClosed && composition != null) {
-            invalidate()
+            snapshotInvalidationTracker.hasInvalidations ||
+            hasPostponedInvalidations
+        if (hasPendingDraws && !isClosed && composition != null) {
+            if (isInvalidationDisabled) {
+                hasPostponedInvalidations = true
```

Comment:
The point of postponing invalidation is to not fire an invalidation if it isn't needed (see the failed tests).

For example, we don't need to fire invalidation in this case:
```
render
  recomposition
    change state -> updateInvalidations -> skip firing
  layout
    read the state -> reset firing, as we don't need a new `render`
```

Why do you need this change?

Reply:
Reverted here.